pytorch_lightning/overrides/base.py

```diff
@@ -19,28 +21,70 @@
 from pytorch_lightning.utilities.device_dtype_mixin import DeviceDtypeModuleMixin
 
 
-class _LightningModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
+class _LightningPrecisionModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
```

Comment:
so a total noob question but why do we need this wrapping just for 64 bit? Why didn't 16 bit have it until now? Do we need it even if amp support 64 bit?

Reply:
The other precision things like AMP just do their own patching so don't need the wrapper. But they don't support 64 bit as they are mainly about trying to avoid floating point errors wioth low precision, not a problem with 64 bit. We only need the wrapping for 64 bit so that we can cast things correctly but still be pickleable for DDP spawn.